crates/router/src/connector/braintree/transformers.rs

```diff
@@ -255,234 +544,1268 @@ impl From<BraintreePaymentStatus> for enums::AttemptStatus {
             BraintreePaymentStatus::Authorized => Self::Authorized,
             BraintreePaymentStatus::Voided => Self::Voided,
             BraintreePaymentStatus::SubmittedForSettlement
-            | BraintreePaymentStatus::SettlementPending
-            | BraintreePaymentStatus::SettlementConfirmed => Self::Pending,
+            | BraintreePaymentStatus::SettlementPending => Self::Pending,
         }
     }
 }
 
-impl<F, T>
-    TryFrom<types::ResponseRouterData<F, BraintreePaymentsResponse, T, types::PaymentsResponseData>>
-    for types::RouterData<F, T, types::PaymentsResponseData>
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            BraintreePaymentsResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsAuthorizeData, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
         item: types::ResponseRouterData<
             F,
             BraintreePaymentsResponse,
```

Comment:
What is the difference between `BraintreePaymentsResponse` and `BraintreeAuthResponse`

Reply:
BraintreePaymentsResponse : Charge
BraintreeAuthResponse: Auth only